tucan/io.py

```diff
@@ -51,8 +51,32 @@ def graph_from_tucan(tucan: str) -> nx.Graph:
 
 
 def _split_into_tokenized_lines(string: str) -> List[List[str]]:
-    lines = [line.rstrip().split(" ") for line in string.splitlines()]
-    return [[value for value in line if value != ""] for line in lines]
+    lines = string.splitlines()
+    lines = _concat_lines_with_dash(lines)
+    split_lines = [line.rstrip().split(" ") for line in lines]
+    return [[value for value in line if value != ""] for line in split_lines]
+
+
+def _concat_lines_with_dash(lines: List[str]) -> List[str]:
+    result = list()
+
+    while (length := len(lines)) > 1:
+        curr_line = lines.pop()
+        prev_line = lines[length - 2]
+
+        if prev_line.endswith("-") and prev_line.startswith("M  V30 "):
+            if curr_line.startswith("M  V30 "):
+                lines[length - 2] = prev_line[0:-1] + curr_line[7:]
+            else:
+                raise MolfileParserException(
+                    f'Invalid concatenation of lines "{prev_line}" and "{curr_line}"'
+                )
+        else:
+            result.append(curr_line)
+
+    result.append(lines[0])
+    result.reverse()
+    return result
```

Comment:
Maybe we could flatten the logic / remove some nesting?

```Python
def _concat_lines_with_dash(lines: List[str]) -> List[str]:
    final_lines = []
    lines = deque(lines)

    while lines:
        curr_line = lines.popleft()

        if not lines:
            final_lines.append(curr_line)
            break

        if not (curr_line.startswith("M  V30 ") and curr_line.endswith("-")):
            final_lines.append(curr_line)
            continue

        next_line = lines.popleft()
        if not next_line.startswith("M  V30 "):
            raise MolfileParserException(
                f'Invalid concatenation of lines "{curr_line}" and "{next_line}"'
            )

        lines.appendleft(curr_line[0:-1] + next_line[7:])

    return final_lines
```
Using the [deque](https://realpython.com/python-deque/) should also be (marginally) more efficient compared to popping and inserting on a list.